fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergCatalog.java

```diff
@@ -238,20 +239,46 @@ default List<Pair<List<Object>, Long>> getSamples() {
         return new ArrayList<>();
     }
 
+    default Map<String, Partition> getPartitionsByDataFiles(String dbName, String tableName, ExecutorService executorService,  Map<String, Partition> partitionMap) {
+        org.apache.iceberg.Table icebergTable = getTable(dbName, tableName);
+
+        if (icebergTable.specs().values().stream().allMatch(PartitionSpec::isUnpartitioned)) {
+            return partitionMap;
+        }
+
+        TableScan tableScan = icebergTable.newScan().planWith(executorService);
+        try (CloseableIterable<FileScanTask> fileScanTaskIterable = tableScan.planFiles();
+                CloseableIterator<FileScanTask> fileScanTaskIterator = fileScanTaskIterable.iterator()) {
+
+            while (fileScanTaskIterator.hasNext()) {
+                FileScanTask scanTask = fileScanTaskIterator.next();
+                StructLike partition = scanTask.file().partition();
+                String partitionName = convertIcebergPartitionToPartitionName(scanTask.spec(), partition);
+                long lastUpdated = -1;
```

Comment:
why this lastUpdated is always -1? it should be a timestamp value which means the last time this partition is updated.

Reply:
There is no good way to get the last modified time of data files. Can we just assign a latest time for each partition?